Add initial structure and examples

RedbackThomson opened the pull request:
Description of changes:
Adds a repository README and directory structure for the examples provided by Emin.

Co-authored-by:  Emin Alemdar <77338109+eminalemdar@users.noreply.github.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Commit:
Add initial structure and examples

Co-authored-by:  Emin Alemdar <77338109+eminalemdar@users.noreply.github.com>

vijtrip2 reviewed (approved):
BEAUTIFULLLLLL 🙌

eminalemdar reviewed (approved):
Brilliant! thanks @RedbackThomson

jaypipes commented on resources/dynamodb/dynamodb-backup.yaml:
```diff
@@ -0,0 +1,7 @@
+apiVersion: dynamodb.services.k8s.aws/v1alpha1
+kind: Backup
+metadata:
+  name: $BACKUP_NAME #Change the Backup Name
```
Do we really need all the comments saying to `Change the XXX`? I would think the wording above to modify any property identified by a dollar sign and all caps is good enough.

eminalemdar replied:
Thanks for the comment @jaypipes ! I've removed the comments from all of the YAML files in the original repository @RedbackThomson you can check it out.

jaypipes commented on resources/dynamodb/dynamodb-backup.yaml:
```diff
@@ -0,0 +1,7 @@
+apiVersion: dynamodb.services.k8s.aws/v1alpha1
```
I would name the file `resources/dynamodb/backup.yaml` or even `resources/dynamodb/v1alpha1/backup.yaml` instead of repeating "dynamo-" in the filename.

That way, you can, potentially, read in these example files in the official Reference API docs and will have the resource name and API version as part of the filepath...

eminalemdar replied:
I haven't updated the folder structure in the original repository but that's a good idea @jaypipes thanks!

jaypipes commented on resources/ec2/dhcp-config.yaml:
```diff
@@ -0,0 +1,14 @@
+apiVersion: ec2.services.k8s.aws/v1alpha1
```
recommend naming the file `resources/ec2/v1alpha1/dhcpoptions.yaml` to ensure you name the file the same as the resource Kind.

eminalemdar replied:
Thanks for the comment @jaypipes ! I've changed the filename to `dhcpoptions.yaml` in the original repository @RedbackThomson you can check it out.

jaypipes commented on resources/s3/bucket_with_encryiption.yaml:
```diff
@@ -0,0 +1,11 @@
+apiVersion: s3.services.k8s.aws/v1alpha1
```
recommend naming the file `resources/s3/v1alpha1/bucket-with-encryption.yaml` (note: spelling error on "encryption" and use hyphens instead of underscores to standardize *variants* using the `-with-` moniker.

eminalemdar replied:
Thanks for the comment @jaypipes ! I've changed the filename to `bucket-with-encryption.yaml` in the original repository @RedbackThomson you can check it out.

Also, I've updated all the filenames with hyphens instead of underscores.

jaypipes reviewed (changes requested):
I'm thinking ahead about how these examples might be useful linked into the official API reference and have suggested some changes to the filenames to standardize the format of them accordingly.

I'm thinking long-term, the API Reference could have an "Examples" tab under each CRD that would link to the variants stored in this repository.

RedbackThomson commented:
> I'm thinking ahead about how these examples might be useful linked into the official API reference and have suggested some changes to the filenames to standardize the format of them accordingly.
> 
> 
> 
> I'm thinking long-term, the API Reference could have an "Examples" tab under each CRD that would link to the variants stored in this repository.

I also wanted to add a table of contents to each resource directory with a brief description of every example

eminalemdar commented:
> I also wanted to add a table of contents to each resource directory with a brief description of every example

That's a great idea! I have simple README files under every Service Controller folder in the original repository but of course those can be improved.

Commit:
Restructure resources by version

Commit:
Add Terraform scripts from remote

RedbackThomson commented:
@eminalemdar I have just rebased the pull request against your latest commits, with a few changes:
- I've added `v1alpha1` directories to each of the service controller directories (as per Jay's comment)
- I've removed any duplication of the service name in the resource file names
- I added the Terraform scripts to `scripts/terraform` and modified the scripts
  - Ensured the new path worked appropriately
  - Updated the Helm installation approach to use proper OCI tooling (no more downloading into a temp file)

jkatz commented on resources/rds/v1alpha1/db-instance.yaml:
```diff
@@ -0,0 +1,27 @@
+  dbInstanceIdentifier: $DB_INSTANCE_ID 
+  dbName: $DB_NAME 
+  dbSubnetGroupName: $DB_SUBNET_GROUP_NAME 
+  dbSecurityGroups:
+  - $DB_SECURITY_GROUP_NAME 
```
This was removed from the CRD: https://github.com/aws-controllers-k8s/rds-controller/pull/80

jkatz commented on resources/rds/v1alpha1/db-security-group.yaml:
```diff
@@ -0,0 +1,8 @@
+---
```
This was removed from the CRD: https://github.com/aws-controllers-k8s/rds-controller/pull/80

Commit:
Remove outdated RDS CRDs

Commit:
Fix TF module path

jaypipes commented on resources/ec2/v1alpha1/elastic-ip.yaml:
```diff
@@ -0,0 +1,11 @@
+---
```
this filename should be `resources/ec2/v1alpha1/elasticipaddress.yaml` to match the resource name :)

jaypipes commented on resources/ec2/v1alpha1/igw.yaml:
```diff
@@ -0,0 +1,7 @@
+---
```
this filename should be `resources/ec2/v1alpha1/internetgateway.yaml` to match the resource name

jaypipes commented on resources/dynamodb/v1alpha1/global-table.yaml:
```diff
@@ -0,0 +1,9 @@
+---
```
this filename should be `resources/dynamodb/v1alpha1/globaltable.yaml` to match the resource name

jaypipes commented on resources/ec2/v1alpha1/nat-gateway.yaml:
```diff
@@ -0,0 +1,14 @@
+---
```
this filename should be `resources/ec2/v1alpha1/natgateway.yaml` to match the resource name

jaypipes commented on resources/ec2/v1alpha1/route-table.yaml:
```diff
@@ -0,0 +1,17 @@
+---
```
this filename should be `resources/ec2/v1alpha1/routetable.yaml`

jaypipes commented on resources/ec2/v1alpha1/security-group.yaml:
```diff
@@ -0,0 +1,14 @@
+---
```
this filename should be `resources/ec2/v1alpha1/securitygroup.yaml`

jaypipes commented on resources/ec2/v1alpha1/transit-gateway.yaml:
```diff
@@ -0,0 +1,12 @@
+---
```
this filename should be `resources/ec2/v1alpha1/transitgateway.yaml`

jaypipes commented on resources/ec2/v1alpha1/vpc-endpoint.yaml:
```diff
@@ -0,0 +1,19 @@
+---
```
this filename should be `resources/ec2/v1alpha1/vpcendpoint.yaml`

jaypipes commented on resources/ecr/v1alpha1/simple-repository.yaml:
```diff
@@ -0,0 +1,7 @@
+---
```
this filename should be `resources/ecr/v1alpha1/repository.yaml`

jaypipes commented on resources/eks/v1alpha1/cluster-addons.yaml:
```diff
@@ -0,0 +1,10 @@
+---
```
this filename should be `resources/eks/v1alpha1/addon.yaml`

jaypipes commented on resources/eks/v1alpha1/fargate-profile.yaml:
```diff
@@ -0,0 +1,15 @@
+---
```
this filename should be `resources/eks/v1alpha1/fargateprofile.yaml`

jaypipes commented on resources/eks/v1alpha1/node-group.yaml:
```diff
@@ -0,0 +1,23 @@
+---
```
this filename should be `resources/eks/v1alpha1/nodegroup.yaml`

jaypipes commented on resources/eks/v1alpha1/simple-cluster.yaml:
```diff
@@ -0,0 +1,15 @@
+---
```
this filename should be `resources/eks/v1alpha1/cluster.yaml`

jaypipes commented on resources/emrcontainers/v1alpha1/virtual-cluster.yaml:
```diff
@@ -0,0 +1,16 @@
+---
```
this filename should be `resources/emrcontainers/v1alpha1/virtualcluster.yaml`

jaypipes commented on resources/kms/v1alpha1/custom-key-store.yaml:
```diff
@@ -0,0 +1,10 @@
+---
```
this filename should be `resources/kms/v1alpha1/customkeystore.yaml`

jaypipes commented on resources/kms/v1alpha1/key-alias.yaml:
```diff
@@ -0,0 +1,8 @@
+---
```
this filename should be `resources/kms/v1alpha1/alias.yaml`

jaypipes commented on resources/rds/v1alpha1/db-cluster-parameter-group.yaml:
```diff
@@ -0,0 +1,25 @@
+---
```
this filename should be `resources/rds/v1alpha1/dbclusterparametergroup.yaml`

jaypipes commented on resources/rds/v1alpha1/db-cluster.yaml:
```diff
@@ -0,0 +1,30 @@
+---
```
this filename should be `resources/rds/v1alpha1/dbcluster.yaml`

jaypipes commented on resources/rds/v1alpha1/db-instance.yaml:
```diff
@@ -0,0 +1,25 @@
+---
```
this filename should be `resources/rds/v1alpha1/dbinstance.yaml`

jaypipes commented on resources/rds/v1alpha1/db-parameter-group.yaml:
```diff
@@ -0,0 +1,25 @@
+---
```
this filename should be `resources/rds/v1alpha1/dbparametergroup.yaml`

jaypipes commented on resources/rds/v1alpha1/db-subnet-group.yaml:
```diff
@@ -0,0 +1,14 @@
+---
```
this filename should be `resources/rds/v1alpha1/dbsubnetgroup.yaml`

jaypipes commented on resources/s3/v1alpha1/simple-bucket.yaml:
```diff
@@ -0,0 +1,7 @@
+---
```
this filename should be `resources/s3/v1alpha1/bucket.yaml`

Commit:
Match resource name with resource kind

RedbackThomson commented:
Thanks Jay lol I think I got the point. The pattern for the file name should be `{CRD Kind}[-{additional context}].yaml`

jaypipes commented:
> Thanks Jay lol I think I got the point. The pattern for the file name should be `{CRD Kind}[-{additional context}].yaml`

sorry man!

jaypipes reviewed (approved):
:tada: :)

eminalemdar commented:
> @eminalemdar I have just rebased the pull request against your latest commits, with a few changes:
> 
> * I've added `v1alpha1` directories to each of the service controller directories (as per Jay's comment)
> * I've removed any duplication of the service name in the resource file names
> * I added the Terraform scripts to `scripts/terraform` and modified the scripts
>   
>   * Ensured the new path worked appropriately
>   * Updated the Helm installation approach to use proper OCI tooling (no more downloading into a temp file)

Thanks a lot @RedbackThomson ! I can/will update the original repository for simplification
